src/main/java/randoop/reflection/OperationExtractor.java

```diff
@@ -152,6 +182,13 @@ public void visit(Constructor<?> constructor) {
     TypedClassOperation operation = instantiateTypes(TypedOperation.forConstructor(constructor));
     checkSubTypes(operation);
     if (!omitPredicate.shouldOmit(operation)) {
+      if (operationSpecifications != null) {
+        OperationConditions conditions =
+            operationSpecifications.getOperationConditions(constructor);
+        if (!conditions.isEmpty()) {
```

Comment:
I think you should do this unconditionally, to avoid expensive re-computation of empty conditions.